src/meta/processors/zone/AddHostsProcessor.cpp

```diff
@@ -50,6 +50,22 @@ void AddHostsProcessor::process(const cpp2::AddHostsReq& req) {
     spaceIter->next();
   }
   for (auto& host : hosts) {
+    auto hostKey = MetaKeyUtils::hostKey(host.host, host.port);
+    auto hostRes = doGet(hostKey);
+    // check if the host is storage, if it is not storage, it should be found in host table not
+    // expired, so it can't be added
+    if (nebula::ok(hostRes)) {
+      auto now = time::WallClock::fastNowInMilliSec();
+      auto hostInfo = HostInfo::decode(nebula::value(hostRes));
+      if (hostInfo.role_ != cpp2::HostRole::STORAGE &&
+          now - hostInfo.lastHBTimeInMilliSec_ <
+              FLAGS_heartbeat_interval_secs * FLAGS_expired_time_factor * 1000) {
```

Comment:
u also need to handle the host which is not storage and is expired.

Reply:
what about add the graphd host ?

Reply:
if it's not expired, it can't be added. but if it's expired, it can be added, because it may be changed to storage

Reply:
yes, if we don't allow to add a expired graphd, it would never be added even if user change it to storage